File: pset1/cash/cash.c
#include<stdio.h>
#include<cs50.h>
#include<math.h>

void money(int c);

int n, cents;

int main(void)
{
    float change;

    do
    {

        change = get_float("Change owed:");

    }

    while (change < 0);

    cents = round(change * 100);

    money(25);

    int a = n;

    money(10);

    int b = n;

    money(5);

    int c = n;

    money(1);

    int d = n;

    printf("%i\n", a + b + c + d);

}

void money(int c)
{
    n = 0;

    while (c * n <= cents)
    {
        n++;
    }

    n--;

    cents = cents - c * n;

}

File: pset6/cash/cash.py
from cs50 import get_float
while True:
    change = get_float("Change owned: ")
    if change > 0.000:
        break

cents = change * 100
n = 0
while (25 * n) <= cents:
    n += 1

n -= 1

cents -= 25 * n

a = n

n = 0

while (10 * n) <= cents:
    n += 1

n -= 1
cents -= 10 * n
b = n

n = 0

while (5 * n) <= cents:
    n += 1

n -= 1

cents -= 5 * n

c = n

n = 0

while (1 * n) <= cents:
    n += 1

n -= 1

cents -= 1 * n

d = n

t = a + b + c + d

print(t)